src/engine.cc

```diff
@@ -155,11 +157,20 @@ void EngineController::NewGame() {
   cache_.Clear();
   search_.reset();
   tree_.reset();
+  current_position_ = optional<CurrentPosition>();
```

Comment:
It's better to add reset()
http://en.cppreference.com/w/cpp/utility/optional/reset

to our poor-man's std::optional and use it here.